.gitmodules

```diff
@@ -0,0 +1,3 @@
+[submodule "flexdll"]
```

Comment:
That is indeed correct - after doing git clone you then have to say either
```sh
git submodule init
git submodule update
```
or you can do it in one with `git submodule update --init`.
It is more convenient if building from a git clone as it's a more convenient command than wget/curl followed by tar. It also, from a source control perspective, formally declares the link between the two projects (i.e. where the library comes from).
All git then stores in a commit in the OCaml repository is the SHA of the commit in the FlexDLL repository which should be checked out (this is what `git submodule update` does - and you have to run it after any `git checkout` if the commit you checked out specifies a different SHA from the one presently checked out). That also means that if, say, there's a fix to FlexDLL which *must* be used (e.g. a new minimum version) then you can formally declare that in the OCaml repository by specifying a different SHA.
Note that by design, what is stored is the SHA and not the branch, so when new features are added to FlexDLL, at some point the OCaml pointer has to be moved forwards - this is fairly simple:
```sh
# From OCaml git clone root directory, with the submodule initialised
cd flexdll
git fetch upstream
git merge upstream/master # or whatever
cd ..
git add flexdll
git commit -m "OCaml now requires FlexDLL > 0.34"
```
Submodules get a lot of bad press because in many instances they're used in contexts where you **must** check out the submodule in order to use the main project, and the need to remember `git submodule update` all the time is tedious. You also have to be careful if you're the kind of git user who uses `git add -u` (or even worse `git add -A`) without checking and understanding the output of `git status` (because if you forgot to `git submodule update` then your `git add -u` will add the wrong SHA for the submodule to your commit). However, none of this should matter as I think more normally all development on the OCaml sources will take place without initialising the submodule (because you shouldn't need it)
I've used it here because I think that it's *exactly* what's wanted!